### PR TITLE
feat: catalog fetch via entity; remove associations

### PR DESCRIPTION
**Pull request by dbouwman**
1. Description:
- **BREAKING** remove `IWithCatalogs` interface and any implementation in Entites
- update the internals of the Catalog.contains functions so that, when necessary, they fetch the catalog via the Hub Entity, thereby assuring that any migrations are applied and abstracting the persistence structure from the Catalog related functions.
- add `getEntityTypeFromHubEntityType` helper
- BREAKING: remove the long deprecated "Association" functions, types and behavior interfaces
- BREAKING: remove the long deprecated `getEntityDefaultWorkspacePane` which was never used, and is implemented in a util in hub components

1. Instructions for testing: run da tests

1. Closes Issues: #<number> (if appropriate)

1. [ ] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [ ] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

**Review by jordantsanz — COMMENTED** [review]
Awesome work with all of this, things look good to me -- just one minor comment

Only thing to add is that events & discussions also seem to have `catalogs` on their defaults -- is this something we'd also want to remove now?

File: packages/common/e2e/catalog-collection-qa.e2e.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { Collection } from "../src/search/Collection";
 import Artifactory from "./helpers/Artifactory";
 import config from "./helpers/config";
 
-describe("catalog and collection e2e:", () => {
+fdescribe("catalog and collection e2e:", () => {
```
Comment (mjuniper) [inline packages/common/e2e/catalog-collection-qa.e2e.ts:6]: I think we want to revert this change, right?

**Review by mjuniper — APPROVED** [review]

**Review by vivzhang — APPROVED** [review]
Looks good to me, just some minor comments.

File: packages/common/e2e/catalog-collection-qa.e2e.ts (hunk 1 above)
Comment (dbouwman) [reply packages/common/e2e/catalog-collection-qa.e2e.ts:6]: it's just in an e2e dev't harness I use to verify that stuff works against the actual api, but I'll remove it

File: packages/common/e2e/hub-sites-contains.e2e.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ fdescribe("HubSite.contains:", () => {
     siteInstance = await HubSite.fetch(siteItemId, context);
   });
 
-  fdescribe("deepCatalogContains", () => {
+  describe("deepCatalogContains", () => {
```
[the remark this answers is not in the document]
Comment (dbouwman) [reply packages/common/e2e/hub-sites-contains.e2e.ts:54]: I'll update these, but as I noted to Mike, these are more "dev harnesses" than e2e's so leaving focus is fine, and some of these e2e's no longer work b/c the fixture items/groups etc have been deleted. :)

File: packages/common/src/search/_internal/getEntityTypeFromType.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import { get } from "http";
+import { HubEntityType } from "../../core";
```
[the remark this answers is not in the document]
Comment (dbouwman) [reply packages/common/src/search/_internal/getEntityTypeFromType.ts:2]: zomg - thanks for catching that!

File: packages/common/src/core/catalogContains.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+ * @param options
+ */
+export async function catalogContains(
+  identifier: string,
+  catalog: IHubCatalog,
+  context: IArcGISContext,
+  options?: IContainsOptions
+): Promise<IContainsResponse> {
+  // Ensure we have options as an object
+  options = options || {};
```
[the remark this answers is not in the document]
Comment (dbouwman) [reply packages/common/src/core/catalogContains.ts:27]: image

so - I don't think you can assign a default to an optional arg in typescript, but having the `= {}` has the effect of making it optional.
